Sources/NIOHTTP2/HTTP2StreamID.swift

```diff
@@ -84,12 +88,27 @@ public class HTTP2StreamID {
         precondition(self.actualStreamID == -1)
         self.actualStreamID = newID
     }
+    
+    /// Obtains a stream-id value with its most significant bit properly cleared, ready to
+    /// be sent on the wire.
+    internal var safeNetworkStreamID: UInt32? {
+        guard self.actualStreamID >= 0 else { return nil }
+        return UInt32(self.actualStreamID) & ~HTTP2StreamID.reservedBitMask
```

Comment:
We should clear this on creation, rather than on emission. We should trap if users attempt to create a stream ID too large.